extension/vector/src/function/query_hnsw_index.cpp

```diff
@@ -167,10 +169,17 @@ static offset_t tableFunc(const TableFuncInput& input, TableFuncOutput& output)
         index->setDefaultUpperEntryPoint(auxInfo.upperEntryPoint);
         index->setDefaultLowerEntryPoint(auxInfo.lowerEntryPoint);
         const auto dimension = ArrayType::getNumElements(getIndexColumnType(bindData->boundInput));
-        auto queryVector = getQueryVector(input.context->clientContext,
-            bindData->boundInput.queryExpression, dimension);
-        localState->result = index->search(input.context->clientContext->getTransaction(),
-            queryVector, localState->searchState);
+        auto indexType = index->getElementType();
+        TypeUtils::visit(indexType, [&]<typename T>(T) {
```

Comment:
I think a better way to use TypeUtils::visit is:
```
TypeUtils::visit(indexType, [&]< vss_element_type T>(T) {
auto queryVector = getQueryVector<T>(input.context->clientContext,
                    bindData->boundInput.queryExpression, index->getElementType(), dimension);
                localState->result = index->search(input.context->clientContext->getTransaction(),
                    queryVector.data(), localState->searchState);
},
[&](auto) {
                KU_UNREACHABLE;
}
```
We should avoid using `if constexpr`